theatre/core/src/propTypes/internals.ts

```diff
@@ -47,12 +48,30 @@ export type ShorthandPropToLonghandProp<P extends IShorthandProp> =
     ? PropTypeConfig_Boolean
     : P extends PropTypeConfig
     ? P
-    : P extends IShorthandCompoundProps
+    : P extends UnknownShorthandCompoundProps
     ? PropTypeConfig_Compound<ShorthandCompoundPropsToLonghandCompoundProps<P>>
     : never
 
+export type ShorthandCompoundPropsToInitialValue<
+  P extends UnknownShorthandCompoundProps,
+> = LonghandCompoundPropsToInitialValue<
+  ShorthandCompoundPropsToLonghandCompoundProps<P>
+>
+
+type LonghandCompoundPropsToInitialValue<P extends IValidCompoundProps> = {
```

Comment:
This could also be written as `ShorthandCompoundPropsToLonghandCompoundProps<P>['valueType']`